### PR TITLE
Update CRDC Data Loader to import the Data Hub released for DELETION

### PR DESCRIPTION
**Pull request by wfy1997**
(1) All changes apply to "delete" mode only, other modes should remain the same.
(2) Validate if id_field or node_type columns is missing in the input_delete_manifest file, if it is missing, abort data deleting
(3) Validate all nodes exist in DB (Neo4j), and give errors otherwise
(4) Disable all other validations. For example, required properties, relationships, etc.

### CODE REVIEW

File: icdc_schema.py [hunk 1]
```diff
@@ -265,7 +266,7 @@ def get_type(self, name):
                         if ITEM_TYPE in prop_desc:
                             item_type = self._get_item_type(prop_desc[ITEM_TYPE])
                             result[ITEM_TYPE] = item_type
-                        elif PROP_ENUM in prop_desc:
+                        if PROP_ENUM in prop_desc:
```
Comment (n2iw) [inline icdc_schema.py:269]: I wonder, what's the reason for this change?

File: icdc_schema.py [hunk 2]
```diff
@@ -37,8 +37,8 @@
 DESCRIPTION = 'Desc'
 
 
-def get_list_values(list_str):
-    return [item.strip() for item in list_str.split(LIST_DELIMITER) if item.strip()]
+def get_list_values(list_str, list_delimiter):
```
Comment (n2iw) [inline icdc_schema.py:40]: I suggest moving **get_list_values** function inside ICDC_Schema class, so that it can use the delimiter in the class/object, instead of asking caller to provide delimiter.

File: icdc_schema.py [hunk 3]
```diff
@@ -512,10 +513,15 @@ def _validate_type(self, model_type, str_value):
                     and not re.match(r'\bltf\b', str_value, re.IGNORECASE)):
                 return False, wrong_type
         elif model_type[PROP_TYPE] == 'Array':
-            for item in get_list_values(str_value):
-                validation_result, error_type = self._validate_type(model_type[ITEM_TYPE], item)
-                if not validation_result:
-                    return False, wrong_type
+            for item in get_list_values(str_value, self.delimiter):
+                if ENUM in model_type[ITEM_TYPE]:
```
Comment (n2iw) [inline icdc_schema.py:517]: Will self._validate_type() work? Why rewrite it here?

File: loader.py [hunk 4]
```diff
@@ -218,6 +218,9 @@ def main(args):
                 if not confirm_deletion('Wipe out entire Neo4j database before loading?'):
                     sys.exit(1)
 
+            if config.loading_mode == DELETE_MODE:
+                config.cheat_mode = True
```
Comment (n2iw) [inline loader.py:222]: Why we need to set cheat_mode here?

File: data_loader.py [hunk 5]
```diff
@@ -191,7 +191,36 @@ def check_files(self, file_list):
...
+                file_encoding = check_encoding(txt)
+                with open(txt, encoding=file_encoding) as in_file:
+                    reader = csv.DictReader(in_file, delimiter='\t')
+                    for org_obj in reader:
+                        obj = self.cleanup_node(org_obj)
+                        id_field = self.schema.get_id_field(obj)
+                        if id_field not in obj.keys():
+                            self.log.error(f'Required id field {id_field} is missing, validation failed')
+                            return False
+                        elif obj[id_field] is None:
```
Comment (n2iw) [inline data_loader.py:207]: Will it handle empty string in id_field?

File: data_loader.py [hunk 6]
```diff
@@ -191,7 +191,36 @@ def check_files(self, file_list):
...
+                            self.log.error(f'Required id field {id_field} is None, validation failed')
+                            return False
+                        if NODE_TYPE not in obj.keys():
+                            self.log.error(f'Required node type field {NODE_TYPE} is missing, validation failed')
+                            return True
+                        elif obj[NODE_TYPE] is None:
```
Comment (n2iw) [inline data_loader.py:213]: Will it handle empty string in type column?

File: icdc_schema.py (hunk 1 above)
Comment (wfy1997) [reply icdc_schema.py:269]: There might be possible that some properties have both ITEM_TYPE and PROP_ENUM

File: icdc_schema.py (hunk 3 above)
Comment (wfy1997) [reply icdc_schema.py:517]: The code is part of the ._validate_type() function

File: loader.py (hunk 4 above)
Comment (wfy1997) [reply loader.py:222]: Because of the "Disable all other validations. For example, required properties, relationships, etc." requirement

File: data_loader.py (hunk 5 above)
Comment (wfy1997) [reply data_loader.py:207]: the script will log the error Invalid data at line 2: "Required property: "file_id" is empty!"! and the error will also be in the data validation result file

File: data_loader.py (hunk 6 above)
Comment (wfy1997) [reply data_loader.py:213]: The script will return 
**ERROR: (ICDC Schema) get_id_field: "type" field is empty
ERROR: (ICDC Schema) get_id_field: "type" field is empty
ERROR: (Data Loader) Invalid data at line 3: "Node type: "" not found in data model"!**
But the script will not store the "type" missing error in the data validation result file

**Review by n2iw — APPROVED** [review]
LGTM
